### PR TITLE
add -init parameter

### PR DESCRIPTION
**Pull request by kovetskiy**

### CODE REVIEW

**Comment by moznion** [comment]
Thank you for your contribution!
Please give me a little time to take a look at this.

**Comment by kovetskiy** [comment]
@moznion have you had a chance to take a look at the PR?

File: internal/constructor/all_args_constructor_generator.go [hunk 1]
```diff
@@ -29,9 +30,24 @@ func (cg *AllArgsConstructorGenerator) Generate() g.Statement {
...
-	return g.NewFunc(
+	retStructure := fmt.Sprintf("&%s{%s}", cg.TypeName, strings.Join(retStructureKeyValues, ","))
+
+	var stmts []g.Statement
+	if cg.InitFunc != "" {
+		stmts = append(stmts, g.NewRawStatementf("r := %s", retStructure))
+		stmts = append(stmts, g.NewRawStatementf("r.%s()", cg.InitFunc))
+		stmts = append(stmts, g.NewReturnStatement("r"))
+	} else {
+		stmts = append(
```
Comment (moznion) [inline internal/constructor/all_args_constructor_generator.go]: [nit] ditto

File: internal/constructor/builder_constructor_generator.go [hunk 2]
```diff
@@ -51,11 +52,25 @@ func (cg *BuilderGenerator) Generate() g.Statement {
 		retStructureKeyValues = append(retStructureKeyValues, fmt.Sprintf("%s: b.%s", field.FieldName, strcase.ToLowerCamel(field.FieldName)))
 	}
 
+	buildResult := fmt.Sprintf("&%s{%s}", cg.TypeName, strings.Join(retStructureKeyValues, ","))
+
+	var buildStmts []g.Statement
+	if cg.InitFunc != "" {
+		buildStmts = append(buildStmts, g.NewRawStatementf("r := %s", buildResult))
```
Comment (moznion) [inline internal/constructor/builder_constructor_generator.go]: [nit] ditto

File: internal/constructor/builder_constructor_generator.go [hunk 3]
```diff
@@ -51,11 +52,25 @@ func (cg *BuilderGenerator) Generate() g.Statement {
...
+		buildStmts = append(buildStmts, g.NewRawStatementf("r.%s()", cg.InitFunc))
+		buildStmts = append(buildStmts, g.NewReturnStatement("r"))
+	} else {
+		buildStmts = append(
+			buildStmts,
```
Comment (moznion) [inline internal/constructor/builder_constructor_generator.go]: [nit] ditto

**Comment by moznion** [comment]
@kovetskiy I'm sorry for my late response. I've just reviewed this pull-request... great. If you feel okay, could you please merge this?

File: internal/constructor/builder_constructor_generator.go (hunk 3 above)
Comment (kovetskiy) [reply internal/constructor/builder_constructor_generator.go]: fixed all of them :small_airplane:

**Comment by moznion** [comment]
Thank you for your contribution!
I've published the new version (v0.1.0) that includes this change: https://github.com/moznion/gonstructor/releases/tag/v0.1.0

**Pull request merged**